File: app/src/main/java/com/example/tipcalculator/MainActivity.java
package com.example.tipcalculator;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioGroup;
import android.widget.Switch;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

     private EditText costOfServiceText;

     private Button calcButton;

     private RadioGroup radioGroup;

     private Switch roundUpSwitch;

     private TextView tipAmountText;

     private double tipAmount;

     private double tipPercentage;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        costOfServiceText = findViewById(R.id.cost_of_service);
        calcButton = findViewById(R.id.calculate_button);
        radioGroup = findViewById(R.id.radioGroup);
        roundUpSwitch = findViewById(R.id.round_up_switch);
        tipAmountText = findViewById(R.id.tip_result);

        calcButton.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View view){
                String cost = costOfServiceText.getText().toString();
                int costAmount = Integer.parseInt(cost);
                int checkedRadioButton = radioGroup.getCheckedRadioButtonId();
                if (checkedRadioButton == R.id.option_twenty_percent){
                    tipPercentage = 0.2;
                } else if (checkedRadioButton == R.id.option_eighteen_percent) {
                    tipPercentage = 0.18;
                } else if (checkedRadioButton == R.id.option_fifteen_percent) {
                    tipPercentage = 0.15;
                }else{
                    tipPercentage = 0.0;
                }

                tipAmount = costAmount * tipPercentage;
                if (roundUpSwitch.isChecked()) {
                    tipAmount = Math.round(tipAmount);
                }

                tipAmountText.setText(""+tipAmount);
            }
        });


    }
}

File: app/src/main/java/com/example/temperatureconverter/MainActivity.kt
package com.example.temperatureconverter

import android.os.Bundle
import android.view.View
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import com.example.temperatureconverter.R
import kotlin.math.roundToInt

class MainActivity : AppCompatActivity() {
    private lateinit var tempInDegree: EditText
    private lateinit var calcButton: Button
    private lateinit var convertOption: RadioGroup
    private lateinit var roundUpSwitch: Switch
    private lateinit var amountText: TextView
    private var convertAmount: Float = 0.0f

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        tempInDegree = findViewById<EditText>(R.id.temp_in_degree)
        calcButton = findViewById<Button>(R.id.calculate_button)
        convertOption = findViewById<RadioGroup>(R.id.radioGroup)
        roundUpSwitch = findViewById<Switch>(R.id.round_up_switch)
        amountText = findViewById<TextView>(R.id.conv_result)

        calcButton.setOnClickListener(View.OnClickListener {
            val cost = tempInDegree.text.toString()
            val costAmount = cost.toFloat()
            when (convertOption.checkedRadioButtonId) {
                R.id.option_Fahrenheit -> convertAmount = (costAmount * 9 / 5 + 32)
                R.id.option_Kelvin -> convertAmount = (costAmount + 273.15).toFloat()
                R.id.option_Rankine -> convertAmount = ((costAmount + 273.15) * 9 / 5).toFloat()
                else -> convertAmount = convertAmount
            }

            //convertAmount = costAmount * convertPercentage;
            if (roundUpSwitch.isChecked) {
                convertAmount = convertAmount.roundToInt().toFloat()
            }
            amountText.text = "Result: ${convertAmount.toString()}"
        })
    }
}